### ISSUE TITLE
Reason for Cross Entropy loss

### ISSUE THREAD

**Issue by pfriesch** [dhgrs/chainer-VQ-VAE#3]
Have you tried a L1/L2 reconstruction loss on the raw data? Or whats the reason you chose a cross entropy loss with mu law? The paper doesn't really say what kind of reconstruction loss is used.

**Comment by dhgrs** [comment]
Sorry for the late reply.

I didn't try L1/L2 reconstruction loss. In the paper, the first term of loss function (reconstruction loss) is log p(x|z_q(x)). So I use cross entropy same as original WaveNet. I think L1/L2 loss doesn't seem to be log p(x|z_q(x)).

I'm trying to find good hyper parameters and now getting good resutls. So please wait for a while.

**Comment by pfriesch** [comment]
I thought the original WaveNet paper only did mu law to reduce the dimensionality and make it tractable. But we already have a reduction here through the embedding. As far as I know, mu law introduces noise/robot like voice. But I will try a different loss and report back! Thank you for your answer! Nice work!

**Comment by dhgrs** [comment]
If you want to predict raw audio without mu-law encoding, descritized mixture of logistics is the solution. It was proposed in [PixelCNN++](https://arxiv.org/abs/1701.05517) and is used in [Parallel WaveNet](https://arxiv.org/abs/1711.10433). I will try it too.
